src/shims/posix/fs.rs

```diff
@@ -73,6 +77,14 @@ impl FileDescriptor for FileHandle {
             Ok(Ok(0))
         }
     }
+
+    fn dup<'tcx>(&mut self) -> InterpResult<'tcx, io::Result<Box<dyn FileDescriptor>>> {
+        self
+            .as_file_handle()
```

Comment:
We know our own type here, which dispatch through that method?

In fact I think you can remove the `InterpResult` here, this should always work from the interpreter side.

Reply:
Oh yeah, missed this while moving this to the trait fn.